eng/pipelines/templates/stages/archetype-java-release.yml

```diff
@@ -0,0 +1,237 @@
+                displayName: 'Use Python 3.6'
+                inputs:
+                  versionSpec: '3.6'
+
+              - script: |
+                  python --version
+                  python eng/versioning/set_versions.py --build-type ${{parameters.SDKType}} --vv --ai ${{artifact.name}} --gi ${{artifact.groupId}}
+                displayName: 'Verify version for release'
+
+              - pwsh: |
+                  $(Build.SourcesDirectory)/eng/versioning/scan_for_unreleased_dependencies.ps1 ${{artifact.groupId}} ${{artifact.name}} $(Build.SourcesDirectory)/sdk/${{parameters.ServiceDirectory}}
+                displayName: "Verify no unreleased dependencies"
+
+
+          - deployment: TagRepository
+            displayName: "Create release tag"
+            condition: and(succeeded(), ne(variables['Skip.TagRepository'], 'true'))
+            environment: github
+            timeoutInMinutes: 5
+            dependsOn: VerifyReleaseVersion
+
+            pool:
+              vmImage: vs2017-win2016
+
+            strategy:
+              runOnce:
+                deploy:
+                  steps:
+                    - checkout: self
+                    - pwsh: |
+                        Get-ChildItem $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
+                        New-Item -Type Directory -Name ${{artifact.safeName}} -Path $(Pipeline.Workspace)
+                        Copy-Item $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.name}}-[0-9]*.[0-9]*.[0-9]* $(Pipeline.Workspace)/${{artifact.safeName}}
+                        Get-ChildItem $(Pipeline.Workspace)/${{artifact.safeName}}
+                      workingDirectory: $(Pipeline.Workspace)
+                      displayName: Stage artifacts
+                      timeoutInMinutes: 5
+                    - template: /eng/common/pipelines/templates/steps/create-tags-and-git-release.yml
+                      parameters:
+                        ArtifactLocation: $(Pipeline.Workspace)/${{artifact.safeName}}
+                        PackageRepository: Maven
+                        ReleaseSha: $(Build.SourceVersion)
+                        RepoId: Azure/azure-sdk-for-java
+
+          - ${{if ne(artifact.options.skipPublishPackage, 'true')}}:
+            - deployment: PublishPackage
+              displayName: "Publish to Maven Central"
+              condition: and(succeeded(), ne(variables['Skip.PublishPackage'], 'true'))
+              environment: maven
+              dependsOn: TagRepository
+
+              pool:
+                vmImage: windows-2019
+
+              strategy:
+                runOnce:
+                  deploy:
+                    steps:
+                      - checkout: azure-sdk-build-tools
+                        path: azure-sdk-build-tools
+                      - template: tools/gpg/gpg.yml@azure-sdk-build-tools
+                      - template: tools/java-publishing/java-publishing.yml@azure-sdk-build-tools
```

Comment:
Is there a reason we don't move the publishing template and scripts into the java repo?

Reply:
We don't want the GPG tooling itself in the repo just yet ;)

Reply:
Agreed but the path to gpg tool could be passed to the java publish template correct?

Reply:
Yep that would work.